eve/io/base.py

```diff
@@ -446,35 +456,40 @@ def _datasource_ex(self, resource, query=None, client_projection=None,
 
         # Only inject the auth_field in the query when not creating new
         # documents.
-        if request and request.method not in ('POST', 'PUT'):
+        if request and request.method != 'POST' and (
+            check_auth_value or force_auth_field_projection
+        ):
             auth_field, request_auth_value = auth_field_and_value(resource)
-            if auth_field and request_auth_value:
-                if query:
-                    # If the auth_field *replaces* a field in the query,
-                    # and the values are /different/, deny the request
-                    # This prevents the auth_field condition from
-                    # overwriting the query (issue #77)
-                    auth_field_in_query = \
-                        self.app.data.query_contains_field(query, auth_field)
-                    if auth_field_in_query and \
+            if auth_field:
+                if request_auth_value and check_auth_value:
+                    if query:
+                        # If the auth_field *replaces* a field in the query,
+                        # and the values are /different/, deny the request
+                        # This prevents the auth_field condition from
+                        # overwriting the query (issue #77)
+                        auth_field_in_query = \
+                            self.app.data.query_contains_field(query,
+                                                               auth_field)
+                        if auth_field_in_query and \
                             self.app.data.get_value_from_query(
                                 query, auth_field) != request_auth_value:
-                        abort(401, description='Incompatible User-Restricted '
-                              'Resource request. '
-                              'Request was for "%s"="%s" but `auth_field` '
-                              'requires "%s"="%s".' % (
-                                  auth_field,
-                                  self.app.data.get_value_from_query(
-                                      query, auth_field),
-                                  auth_field,
-                                  request_auth_value)
-                              )
+                            desc = 'Incompatible User-Restricted Resource ' \
```

Comment:
this might disclose a little too much info to the client. Yes, the desc field is only emitted in debug mode, but you never know someone might leave debug on by mistake.

Reply:
The description string was already present, I didn't change it. I simply restructured this piece of code to fix a Flake8 line too long error. Do you want me to change the description anyway? What should I write?

Reply:
Yes, you are right, it was there already. Let us obfuscate it a little bit, maybe by simply returning the first part of the message (drop the "request was for but xx was excpected")

Reply:
done ;)